examples/file_block_encrypt_example/block_writer.c

```diff
@@ -0,0 +1,141 @@
+
+            // The buffer that will contain multiple key_values and compose a
+            // single block.
+            unsigned char plaintext_block_buf[MAX_PLAINTEXT_BLOCK_SIZE];
+            unsigned int total_block_size = 0;
+
+            if (temp_buf_length != 0) {
+                memcpy(plaintext_block_buf, temp_buf, temp_buf_length);
+                total_block_size += temp_buf_length;
+            }
+
+            // Create random key value buffers.
+            while (total_block_size < MAX_PLAINTEXT_BLOCK_SIZE) {
+                // Create random int between 56~144
+                int plaintext_buf_length =
+                    rand() % (MAX_KEY_VALUE_SIZE + 1 - MIN_KEY_VALUE_SIZE) +
+                    MIN_KEY_VALUE_SIZE;
+                // This buffer contains a single key_value.
+                unsigned char
+                    plaintext_buf[plaintext_buf_length];  // Variable Length
+                                                          // Arrays work from
+                                                          // C99
+
+                // Insert random bytes inside buffer.
+                RAND_bytes(plaintext_buf, plaintext_buf_length);
+
+                // If the block size exceeds MAX_PLAINTEXT_BLOCK_SIZE after
+                // adding the next block, it should be saved to the next block.
+                // The leftover space should be filled with zero padding.
+                if (total_block_size + plaintext_buf_length >
+                    MAX_PLAINTEXT_BLOCK_SIZE) {
+                    // Add zero padding to the end of the plaintext_block_buf.
+                    bzero(plaintext_block_buf + total_block_size,
+                          MAX_PLAINTEXT_BLOCK_SIZE - total_block_size);
+                    printf("Add zero paddings for the leftover %d bytes.\n",
+                           MAX_PLAINTEXT_BLOCK_SIZE - total_block_size);
+                    // Now the total_block_size becomes the
+                    // MAX_PLAINTEXT_BLOCK_SIZE.
+                    total_block_size +=
+                        MAX_PLAINTEXT_BLOCK_SIZE - total_block_size;
+
+                    // Save the created random buffer to put it into the next
+                    // block.
+                    temp_buf_length = plaintext_buf_length;
+                    memcpy(temp_buf, plaintext_buf, temp_buf_length);
+                    break;
+                } else {
+                    memcpy(plaintext_block_buf + total_block_size,
+                           plaintext_buf, plaintext_buf_length);
+                    total_block_size += plaintext_buf_length;
+                }
+            }
+
+            // Save the plaintext block for future comparison.
+            fwrite(plaintext_block_buf, total_block_size, 1, plaintext_fp);
+            plaintext_file_metadata[i].block_metadata[j].length =
+                total_block_size;
+
+            unsigned int encrypted_length;
+            unsigned char *encrypted;
```

Comment:
Whenever possible, I recommend initializing pointers to NULL.

Reply:
Fixed here. 087f0c332796a0385d44d2e07028d6f9a753d25b